core/corerepo/gc.go

```diff
@@ -66,3 +71,110 @@ func GarbageCollectAsync(n *core.IpfsNode, ctx context.Context) (<-chan *KeyRemo
 	}()
 	return output, nil
 }
+
+func PeriodicGC(ctx context.Context, node *core.IpfsNode) error {
+	r := node.Repo
+	cfg := r.Config()
+	if cfg.Datastore.GCPeriod == "" {
+		r.SetConfigKey("Datastore.GCPeriod", "1h")
+		cfg.Datastore.GCPeriod = "1h"
+	}
+
+	period, err := time.ParseDuration(cfg.Datastore.GCPeriod)
+	if err != nil {
+		return err
+	}
+	if int64(period) == 0 {
+		// if duration is 0, it means GC is disabled.
+		return nil
+	}
+
+	storageMax, storageGC, slackGB, err := getGCParam(node.Repo)
```

Comment:
in cases like this we can make structs:

```go
type GC struct {
  Node core.IpfsNode
  Repo repo.Repo
  StorageMax uint64
  StorageGC uint64
  SlackGB uint64
}

func NewGC(n core.IpfsNode) (*GC, error) {
  m, gc, s, err := getGCParam(n.Repo)
  if err != nil {
    return nil, err
  }
  return &GC{
    Node: n,
    Repo: n.Repo,
    StorageMax: m,
    StorageGC: gc,
    SlackGB: s,
  }
}

func (gc *GC) PeriodicGC(ctx context.Context) { ... }
func (gc *GC) ConditionalGC(ctx context.Context, offset int) { ... }
func (gc *GC) maybeGC(ctx context.Context, offset int) { ... }
```
not a big deal for now, i can merge this without and we can improve it later.